lib/network/webserver.js

```diff
@@ -0,0 +1,348 @@
+'use strict'; // eslint-disable-line
+
+const http = require('http');
+const https = require('https');
+const assert = require('assert');
+const Logger = require('werelogs').Logger;
+const dhparam = require('../https/dh2048').dhparam;
+const ciphers = require('../https/ciphers').ciphers;
+const errors = require('../errors');
+
+class WebServer {
+
+    /**
+     * @constructor
+     *
+     * @param {number} port - Port to listen into
+     * @param {werelogs.Logger} logger - Logger object
+     */
+    constructor(port, logger) {
+        assert.strictEqual(typeof port, 'number', 'Port must be a number');
+        assert.strictEqual(logger instanceof Logger, true,
+            'Logger must be an instance of werelogs');
+        this._keepAlive = false;
+        this._noDelay = true;
+        this._cbOnListening = () => {};
+        this._cbOnRequest = null;
+        this._cbOnError = () => false;
+        this._cbOnStop = () => {};
+        this._https = {
+            ciphers,
+            dhparam,
+            cert: null,
+            key: null,
+            ca: null,
+            requestCert: false,
+            rejectUnauthorized: true,
+        };
+        this._port = port;
+        this._server = null;
+        this._logger = logger;
+    }
+
+    /**
+     * Setter to noDelay, this disable the nagle tcp algorithm, reducing
+     * latency for each request
+     *
+     * @param {boolean} value - { true: Disable, false: Enable }
+     * @return {WebServer} itself
+     */
+    setNoDelay(value) {
+        this._noDelay = value;
+        return this;
+    }
+
+    /**
+     * Getter to access to the http/https server
+     *
+     * @return {http.Server|https.Server} http/https server
+     */
+    get server() {
+        return this._server;
+    }
+
+    /**
+     * Getter to access to the current authority certificate
+     *
+     * @return {string} Authority certificate
+     */
+    get authorityCertificate() {
```

Comment:
It's basically a property of the class, so i don't think there is an impact on performances :)